solvebio/query.py

```diff
@@ -1088,19 +1132,31 @@ def execute(self, offset=0, **query):
 
         _params.update(
             offset=self._page_offset,
-            limit=self._limit
+            limit=min(self._page_size, self._limit)
         )
 
         logger.debug('executing query. from/limit: %6d/%d' %
                      (_params['offset'], _params['limit']))
 
         # If the request results in a SolveError (ie bad filter) set the error.
         try:
-            self._response = self._client.get(self._data_url, _params)
+            self._response = self._client.post(self._data_url, _params)
         except SolveError as e:
             self._error = e
             raise
 
-        logger.debug('query response took: %(took)d ms, total: %(total)d'
-                     % self._response)
+        logger.debug('query response took: {} ms, total: {}'.
+                     format(self._response['took'],
+                            'nan' if self._response['total'] is None else self._response['total']))
```

Comment:
It is probably ok to just print the contents of [total] as-is, since it is a debug log.

Reply:
Done.